app/forms/supplementary/DocumentsProduced.scala

```diff
@@ -29,32 +28,27 @@ case class DocumentsProduced(
   documentStatus: Option[String],
   documentStatusReason: Option[String],
   documentQuantity: Option[String]
-) extends MetadataPropertiesConvertable {
+) {
+  implicit val writes = Json.writes[DocumentsProduced]
+  
+  def isDefined: Boolean =
+    List(documentTypeCode, documentIdentifier, documentPart, documentStatus, documentStatusReason, documentQuantity)
+      .exists(_.isDefined)
 
-  override def toMetadataProperties(): Map[String, String] =
-    Map(
-      "declaration.goodsShipment.governmentAgencyGoodsItems[0].additionalDocuments[0].categoryCode" ->
-        documentTypeCode.flatMap(_.headOption).fold("")(_.toString),
-      "declaration.goodsShipment.governmentAgencyGoodsItems[0].additionalDocuments[0].typeCode" ->
-        documentTypeCode.map(_.drop(1).toString).getOrElse(""),
-      "declaration.goodsShipment.governmentAgencyGoodsItems[0].additionalDocuments[0].id" ->
-        (documentIdentifier.getOrElse("") + documentPart.getOrElse("")),
-      "declaration.goodsShipment.governmentAgencyGoodsItems[0].additionalDocuments[0].lpcoExemptionCode" ->
-        documentStatus.getOrElse(""),
-      "declaration.goodsShipment.governmentAgencyGoodsItems[0].additionalDocuments[0].name" ->
-        documentStatusReason.getOrElse(""),
-      "declaration.goodsShipment.governmentAgencyGoodsItems[0].additionalDocuments[0].writeOff.quantity" ->
-        documentQuantity.getOrElse("")
-    )
+  def toJson: JsValue = Json.toJson(this)
 }
 
 object DocumentsProduced {
+
   implicit val format = Json.format[DocumentsProduced]
 
+  implicit val reads = Json.reads[DocumentsProduced]
```

Comment:
Hmm, why do you need `implicit val reads` when you have above `implicit val format`? Format contains reads and writes. It can be some corner case that scala cannot use only format, but I'm not sure about it